### PR TITLE
fixed 5592 to prevent jvm crash when kafka brokers are not reachable

### PR DESCRIPTION
**Pull request by guptaneeru**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. Note that there should always be a related issue alongside a PR as that enables appropriate discussion.

Fixes # (issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Consider any more risky areas that you need advice on.

# Any additional notes for reviewers?

### CODE REVIEW

**Comment by guptaneeru** [comment]
@planetf1 can you please take a look here.

File: open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnector.java [hunk 1]
```diff
@@ -420,8 +420,8 @@ int getNumberOfUnprocessedEvents() {
         boolean getRunningBrokers(Properties connectionProperties ) {
 
             boolean found = false;
-            try  (AdminClient adminClient = KafkaAdminClient.create(connectionProperties))
-            {
+            AdminClient adminClient = KafkaAdminClient.create(connectionProperties)
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnector.java:423]: If adminClient is null ie the create fails, we will end up exiting the retry loop.
However I can't see any compelling failures that might lead to this so probably ok.
Are you comfortable with KafkaAdminClient behaviour?

**Comment by planetf1** [comment]
You need to sign off the commit as per the DCO. See https://github.com/apps/dco

**Review by planetf1 — CHANGES_REQUESTED** [review]
- Confirm/update behaviour if KafkaAdminClient.create() fails
- Sign DCO

File: open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnector.java (hunk 1 above)
Comment (guptaneeru) [reply open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnector.java:423]: Handled NPE.

Comment (planetf1) [reply open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnector.java:423]: Thanks

**Review by planetf1 — APPROVED** [review]

**Comment by planetf1** [comment]
Hi @guptaneeru , thanks for making the updates for the now, but there is now a syntax issue that needs resolution before we can merge.

We will be branching this week. this update looks valuable to include in the release.

**Comment by planetf1** [comment]
If you click on 'checks' you should be able to see the details of what failed. You can click each one for more details which should show the exact error.

If it's a code error you should see it when you build locally. I suggest you do a 'mvn clean install' from the top directory so you can fix locally. (We also have a gradle build foot which you can run './gradlew build'

**Comment by planetf1** [comment]
Your error is in /Users/jonesn/IdeaProjects/egeria.maven/open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnector.java at line 423 - the semicolon is missing at the end of this line.

**Comment by planetf1** [comment]
The DCO needs updating.
Also have you done any testing of the changes in both normal and error cases?
You can take the assembly from the build output if it helps.
The fix looks good by inspection but should test some kafka scenario before merging (our FVTs don't currently as they are single-server not cohort based)

**Comment by guptaneeru** [comment]
Nigel, let me think about some ways to manually test it locally. One of the challenge is WKC uses 2.8 and this build might not be compatible with WKC code. I will try.

**Review by planetf1 — CHANGES_REQUESTED** [review]
I think we need one further change. Other than that the code seems to make sense to me having checked hopefully thoroughly. 

Also the DCO is still incomplete - and without this we cannot incorporate your commits.

If you are having trouble I am happy to recode & merge as needed - could you perhaps take a look in the next few days. I'll redo if I don't hear as the issue you raised is valuable to address.

**Comment by planetf1** [comment]
No activity, so closing & replaced with #5684

**Pull request closed without merging**